### PR TITLE
utils-common: io-slices: Unit tests and small fixes

### PR DESCRIPTION
**Pull request by osteffenrh**
This PR renames the `WalkableIoSliceIter::all_aligned_to()` function to `all_lengths_multipleOf()` and unifies the behavior of `all_lengths_multiple_of(0)` corner case across implementations.

A small bug in the `for_each()` implementation for IoSlicesIterChain is fixed.

Unit tests for all relevant implementations of the io_slices traits under utils-common/io_slices is are added.

This PR supersedes #16, #19, and #20.

### CODE REVIEW

**Comment by osteffenrh** [comment]
```diff
@@ -1,13 +1,13 @@
 Filename                      Regions    Missed Regions     Cover   Functions  Missed Functions  Executed       Lines      Missed Lines     Cover    Branches   Missed Branches     Cover
 -----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 alloc.rs                           56                56     0.00%           4                 4     0.00%          32                32     0.00%           0                 0         -
-bitmanip.rs                       199               199     0.00%          19                19     0.00%         124               124     0.00%           0                 0         -
-ct_cmp.rs                         267                 6    97.75%           6                 0   100.00%         115                 2    98.26%           0                 0         -
+bitmanip.rs                       199               189     5.03%          19                17    10.53%         124               117     5.65%           0                 0         -
+ct_cmp.rs                         267                 3    98.88%           6                 0   100.00%         115                 1    99.13%           0                 0         -
 fixed_vec.rs                      590                89    84.92%          50                12    76.00%         400                87    78.25%           0                 0         -
 index_permutation.rs              105               105     0.00%           6                 6     0.00%          71                71     0.00%           0                 0         -
-io_slices.rs                     2453              1299    47.04%         192               164    14.58%        1359               878    35.39%           0                 0         -
+io_slices.rs                     2461               265    89.23%         192                33    82.81%        1367               168    87.71%           0                 0         -
 murmurhash3.rs                    107                28    73.83%           8                 0   100.00%          85                12    85.88%           0                 0         -
 xor.rs                            240                62    74.17%           3                 0   100.00%         108                34    68.52%           0                 0         -
 zeroize.rs                         85                85     0.00%          11                11     0.00%          60                60     0.00%           0                 0         -
 -----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
-TOTAL                            4102              1929    52.97%         299               216    27.76%        2354              1300    44.77%           0                 0         -
+TOTAL                            4110               882    78.54%         299                83    72.24%        2362               582    75.36%           0                 0         -
```

**Review by nicstange — COMMENTED** [review]
This is absolutely awesome!

One minor nit: the new test files are lacking a SPDX header / copyright statement. Care to amend those?

**Comment by osteffenrh** [comment]
v2:
- Add SPDX / copyright headers

**Comment by nicstange** [comment]
I love this PR, thanks a lot for all the work! And apologies for the delay...
